patches/minecraft/net/minecraft/client/gui/GuiScreen.java.patch

```diff
@@ -35,15 +35,21 @@
          {
              GlStateManager.func_179101_C();
              RenderHelper.func_74518_a();
-@@ -418,6 +428,7 @@
+@@ -414,10 +424,13 @@
+ 
+     public void func_175281_b(String p_175281_1_, boolean p_175281_2_)
+     {
++        p_175281_1_=net.minecraftforge.event.ForgeEventFactory.onClientSendMessage(p_175281_1_);
++        if(p_175281_1_.isEmpty())return;
```

Comment:
@williewillus Alright, I'll fix that when I get the chance.

@LexManos Where do you see there being a potential NPE? Also, for reasons as to why, one particularly good argument for it that I was was [EZTouch's comment](https://github.com/MinecraftForge/MinecraftForge/issues/1442#issuecomment-133676386). And yes, it is above command handler, but still within sendChatMessage(String msg, boolean addToChat), immediately before addToChat is checked and used. This allows us to manipulate the message before it gets used in
`this.mc.ingameGUI.getChatGUI().addToSentMessages(msg);`